spec/spree_shared_spec.rb

```diff
@@ -35,12 +36,61 @@
   end
 
   describe 'preferences' do
-    it 'does not interfere' do
-      Spree::Config.logo = "logo/spree_51.png"
-      Apartment::Tenant.switch!("tenant271")
-      expect(Spree::Config.logo).to eq("logo/spree_50.png")
-      Apartment::Tenant.switch!
-      expect(Spree::Config.logo).to eq("logo/spree_51.png")
+
+    shared_examples 'preferences changing' do
+      it 'preferences do not interfere between tenants' do
+        Spree::Config.logo = "logo/spree_51.png"
+        Apartment::Tenant.switch!("tenant271")
+        expect(Spree::Config.logo).to eq("logo/spree_50.png")
+        Apartment::Tenant.switch!
+        expect(Spree::Config.logo).to eq("logo/spree_51.png")
+      end
+    end
+
+    context 'When cache is enabled' do
+      let(:memory_store) do
+        ActiveSupport::Cache.lookup_store(
+            :memory_store,
+            { namespace: -> { Apartment::Tenant.current } }
+        )
+      end
+
+      before do
+        allow(Rails.cache).to receive(:exist?) { |key| memory_store.exist?(key) }
+        allow(Rails.cache).to receive(:read) { |key| memory_store.read(key) }
+        allow(Rails.cache).to receive(:write) { |key, value| memory_store.write(key, value) }
```

Comment:
Can we stub `Rails.cache` itself here, please? I feel It can behave more reliable than to stub selective methods

Reply:
I'll try ;)

Reply:
Spree::Preferences::Store is a singleton and uses internal @cache variable which normally equals to Rails.cache. I cannot stub `Rails.cache` because of that. And because its a singleton the only reasonable way to stub it is to make 
`Spree::Preferences::Store.instance.instance_variable_set(:@cache, memory_store)`